Sources/TranscribeCLI/main.swift

```diff
@@ -3,42 +3,56 @@
+///     SPEAKER: Name of the speaker. Should be ordered by appearance.
 ///
 ///     The resulting Markdown and WebVTT file is saved to the current directory with the
 ///     same basename as the input file.
 
 import Foundation
 import Transcribe
-
-struct TranscribeCLIError: Error {
-    var code: Int
-    var message: String
-
-    static let missingArgument = TranscribeCLIError(code: 1, message: "Usage: \(CommandLine.arguments[0]) FILE")
-    static func argumentIsNotAFile(argument: String) -> TranscribeCLIError {
-        return TranscribeCLIError(code: 2, message: "\(argument) must be a file")
-    }
-}
+import Utility
 
 // MARK: - Main program
 
 do {
-    guard CommandLine.arguments.count == 2 else {
-        throw TranscribeCLIError.missingArgument
+    // setup the argument parser
+    let parser = ArgumentParser(usage: "<options>",
+                                overview: "A Swift parser for output files from automated transcription services.")
+    let rawInputFilename = parser.add(option: "--json", shortName: "-j", kind: PathArgument.self, usage: "Amazon transcribe json file, e.g. './TestFixtures/amazon-transcribe-swift-community-podcast-0001-formatted-short.json'")
+    let rawSpeakerNames = parser.add(option: "--names", shortName: "-n", kind: [String].self, usage: "Array of speaker names, e.g. 'alice bob'")
+    
+    // The first argument is always the executable, drop it
+    let arguments = Array(ProcessInfo.processInfo.arguments.dropFirst())
+    let parsedArguments = try parser.parse(arguments)
+    guard let inputFilename = parsedArguments.get(rawInputFilename),
+        inputFilename.path.asString.lowercased().hasSuffix("json") else {
+        throw ArgumentParserError.invalidValue(argument: "--json", error: ArgumentConversionError.custom("No json file specified"))
     }
-
-    let inputFilename = CommandLine.arguments[1]
-    let inputFile = URL(fileURLWithPath: inputFilename).standardizedFileURL
-    guard let isRegularFile = try inputFile.resourceValues(forKeys: [.isRegularFileKey]).isRegularFile, isRegularFile == true else {
-        throw TranscribeCLIError.argumentIsNotAFile(argument: inputFile.path)
+   
+    // validate json filename
+    let inputFile = URL(fileURLWithPath: inputFilename.path.asString).standardizedFileURL
+    guard let isRegularFile = try? inputFile.resourceValues(forKeys: [.isRegularFileKey]).isRegularFile, isRegularFile == true else {
+        throw ArgumentParserError.invalidValue(argument: "--json", error: ArgumentConversionError.custom("File not found"))
     }
 
-    let transcript = try AmazonTranscribe.Transcript(file: inputFile)
+    // parse the amazon transcription json file
+    var transcript = try AmazonTranscribe.Transcript(file: inputFile)
     let outputDirectory = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
 
+    // Set speaker names
+    if let speakerNames = parsedArguments.get(rawSpeakerNames),
+        speakerNames.count == transcript.speakers.count {
```

Comment:
Related to my comment above, I'm not sure this should be an error. The Amazon transcript file includes 4 speakers although there were only 3 in the recording. You could argue that it's important not to ignore this, but I don't think the conversion should fail necessarily.

Reply:
I also think passing any speaker names on the command line should be optional. At the moment the `--names` option is required, but it should be possible to run the CLI tool without specifying any speaker name mappings.

Reply:
You are right, in my opionion this might be a warning!? So that the last speaker gets the "spk_3" label and there will be some output like:
```
Found 4 speakers in the transcription, but only 3 names were specified:
spk_0: John Sundell
spk_1: Garric Nahapetian
spk_2: Chris Lattner
```
What do you think?

Reply:
Yes, a warning sounds good. I still like the idea to be able to omit the `--names` param completely, and then the tool shouldn't print any warning at all – after all, the source file could already contain the correct speaker names.